fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -52,39 +63,71 @@ public IcebergMetadata(Map<String, String> properties) {
+     */
+    private static CacheBuilder<Object, Object> newCacheBuilder(long maximumSize) {
+        CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+        cacheBuilder.expireAfterWrite(Config.hive_meta_cache_ttl_s, SECONDS);
+        cacheBuilder.maximumSize(maximumSize);
+        return cacheBuilder;
     }
 
     @Override
     public List<String> listDbNames() throws DdlException {
         return icebergCatalog.listAllDatabases();
     }
 
+    @Override
+    public List<String> listTableNames(String dbName) throws DdlException {
+        List<TableIdentifier> tableIdentifiers = icebergCatalog.listTables(Namespace.of(dbName));
+        return tableIdentifiers.stream().map(TableIdentifier::name).collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    public Table loadTable(String dbName, String tblName) throws DdlException {
+        org.apache.iceberg.Table icebergTable
+                = icebergCatalog.loadTable(IcebergUtil.getIcebergTableIdentifier(dbName, tblName));
+        if (IcebergCatalogType.fromString(catalogType).equals(IcebergCatalogType.CUSTOM_CATALOG)) {
+            return IcebergUtil.convertCustomCatalogToSRTable(icebergTable, catalogImpl, dbName, tblName,
+                    customProperties);
+        }
+        return IcebergUtil.convertHiveCatalogToSRTable(icebergTable, metastoreURI, dbName, tblName);
+    }
+
+    public Database loadDatabase(String dbName) throws TException, InterruptedException {
+        return icebergCatalog.getDB(dbName);
+    }
+
     @Override
     public Database getDb(String dbName) {
         try {
-            return icebergCatalog.getDB(dbName);
-        } catch (InterruptedException | TException e) {
+            return dbCache.get(dbName);
+        } catch (ExecutionException e) {
             LOG.error("Failed to get iceberg database " + dbName, e);
             return null;
         }
     }
 
-    @Override
-    public List<String> listTableNames(String dbName) throws DdlException {
-        List<TableIdentifier> tableIdentifiers = icebergCatalog.listTables(Namespace.of(dbName));
-        return tableIdentifiers.stream().map(TableIdentifier::name).collect(Collectors.toCollection(ArrayList::new));
-    }
-
     @Override
     public Table getTable(String dbName, String tblName) {
         try {
-            org.apache.iceberg.Table icebergTable
-                    = icebergCatalog.loadTable(IcebergUtil.getIcebergTableIdentifier(dbName, tblName));
-            if (IcebergCatalogType.fromString(catalogType).equals(IcebergCatalogType.CUSTOM_CATALOG)) {
-                return IcebergUtil.convertCustomCatalogToSRTable(icebergTable, catalogImpl, dbName, tblName, customProperties);
-            }
-            return IcebergUtil.convertHiveCatalogToSRTable(icebergTable, metastoreURI, dbName, tblName);
-        } catch (DdlException e) {
+            HiveTableName key = new HiveTableName(dbName, tblName);
```

Comment:
IMO, If we cache the IcebergTable, it seems that we couldn't get the newest currentSnapshot when construct scan range of iceberg table.